js/models/entity.js

```diff
@@ -0,0 +1,10 @@
+class Entity{
+    constructor(imagePath, rotation, direction,width,height,mirrorFlippedHorizontal=0){
+        this.imagePath = imagePath;
+        this.rotation = rotation;
+        this.direction = direction;
+        this.width = width;
+        this.height = height;
+        this.mirrorFlippedHorizontal = mirrorFlippedHorizontal;
+    }
+}
```

Comment:
if you want to keep using the array of data, you could create a static method that returns a new Entity:
```javascript
static function fromDataArray(dataArray) {
  var name = dataArray[0];
  var rotation = dataArray[1];
  //etc...
  return new Entity(name, rotation, //etc...
}
```

Reply:
Ok, I'll add this. Eventually I'd like to get away from the array format and have a config file somewhere storing all of the entity information.